pkg/distributor/query.go

```diff
@@ -222,8 +223,9 @@ func mergeExemplarQueryResponses(results []interface{}) *ingester_client.Exempla
 // queryIngesterStream queries the ingesters using the new streaming API.
 func (d *Distributor) queryIngesterStream(ctx context.Context, replicationSet ring.ReplicationSet, req *ingester_client.QueryRequest) (*ingester_client.QueryStreamResponse, error) {
 	var (
-		queryLimiter = limiter.QueryLimiterFromContextWithFallback(ctx)
-		reqStats     = stats.FromContext(ctx)
+		queryLimiter       = limiter.QueryLimiterFromContextWithFallback(ctx)
+		reqStats           = stats.FromContext(ctx)
+		partialDataEnabled = partialdata.FromContext(ctx)
```

Comment:
This feels like an overuse of context. Shouldn't the method signature be changed to pass partitalDataEnabled as a bool?

Reply:
`streamingSelect` calls `q.distributor.QueryStream`, which I thought was a very generic function used by multiple structs.

But re-checking the code, seems like it's onl implemented in one place. I'll make the change.

Reply:
Change made